azure-iot-device/azure/iot/device/common/chainable_exception.py

```diff
@@ -0,0 +1,15 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+
+class ChainableException(Exception):
+    """This exception stores a reference to a previous exception"""
+
+    def __init__(self, message=None, cause=None):
```

Comment:
>__init__ [](start = 8, length = 8)

Since this is sort-of our base class for all exceptions, should we be capturing the stack here?

Reply:
Not sure what you mean by this. I'll follow up with you offline.

---
In reply to: [326734879](https://github.com/Azure/azure-iot-sdk-python/pull/309#discussion_r326734879) [](ancestors = 326734879)